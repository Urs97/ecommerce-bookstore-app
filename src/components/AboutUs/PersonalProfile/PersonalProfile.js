import React from "react";
import "./PersonalProfile.scss";

const PersonalProfile = ({ aboutUsPhoto, name, title, text }) => {
  return (
    <section className="personal-profile">
      {aboutUsPhoto}
      <h3 className="title">{name}</h3>
      <h4 className="title">{title}</h4>
      <p>{text}</p>
    </section>
  );
};

export default PersonalProfile;
