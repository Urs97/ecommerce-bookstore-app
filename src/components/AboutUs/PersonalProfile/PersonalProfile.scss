@import "../../../styles/shared.scss";

.personal-profile {
  padding: 0 5em;

  & img {
    width: 7em;
    height: 7em;
    border: 0.2em solid #ffffff;
    border-radius: 50%;
    object-fit: cover;
    margin-bottom: 1em;
  }

  & h3 {
    font-size: 1.25em;
    margin-bottom: 0.25em;
  }

  & h4 {
    font-size: 1.25em;
    font-weight: normal;
    margin-bottom: 1em;
  }

  & p {
    font-size: 1.25em;
    padding-bottom: 1em;
  }
}
