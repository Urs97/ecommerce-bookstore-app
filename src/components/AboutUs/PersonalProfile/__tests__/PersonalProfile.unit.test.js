import React from 'react';
import {shallow} from 'enzyme';
import PersonalProfile from '../PersonalProfile';
import portraitImg from '../../../../assets/images/testimonial-portrait-1.webp';

describe('PersonalProfile', () => {
    test('renders empty PersonalProfile', () => {
        const wrapper = shallow(
            <PersonalProfile />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders PersonalProfile with props', () => {
        const wrapper = shallow(
            <PersonalProfile 
                imgSrc={portraitImg} 
                name="Larry Jones" 
                title="CEO & Founder" 
                text="Larry Jones is company's founder and longest serving employee"/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});