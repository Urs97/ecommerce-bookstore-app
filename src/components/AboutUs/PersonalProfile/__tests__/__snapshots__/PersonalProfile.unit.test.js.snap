// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalProfile renders PersonalProfile with props 1`] = `
<section
  className="personal-profile"
>
  <img
    alt="portrait"
    src="testimonial-portrait-1.jpg"
  />
  <h3
    className="title"
  >
    Larry Jones
  </h3>
  <h4
    className="title"
  >
    CEO & Founder
  </h4>
  <p>
    Larry Jones is company's founder and longest serving employee
  </p>
</section>
`;

exports[`PersonalProfile renders empty PersonalProfile 1`] = `
<section
  className="personal-profile"
>
  <img
    alt="portrait"
  />
  <h3
    className="title"
  />
  <h4
    className="title"
  />
  <p />
</section>
`;
